src/test/ui/resolve/issue-82825.rs

```diff
@@ -0,0 +1,20 @@
+// check-pass
+
+trait Trait {
+    fn static_call(&self) where Self: Sized;
+    fn maybe_dynamic_call(&self) {
+        unimplemented!("unsupported maybe_dynamic_call");
+    }
+}
+
+impl<T: ?Sized + Trait> Trait for &T {
+    fn static_call(&self) where Self: Sized {
+        (**self).maybe_dynamic_call();
+    }
+}
+
+fn foo(x: &dyn Trait) {
+    x.static_call();
+}
```

Comment:
@b-naber can we add a &mut test here? that was the motivating usecase

also, is there a way we can test that this has actually regressed diagnostics? ci appears to pass just fine

Reply:
Added the test, but we have to declare x as mutable in order for the mutable autoref to work, which might be confusing given that the autoref is implicit.

Reply:
We don't have diagnostics problems in this version (using the existing `illegal_sized_bound` hack.

Reply:
@b-naber oh thats very confusing...is the diagnostic around that good? as in, does it explicitly tell you to add the `mut` before the parameter?

Reply:
Hm not really great, the borrow part is confusing, I don't think we should expect the user to know that an autoref occurs here:

```
error[E0596]: cannot borrow `x` as mutable, as it is not declared as mutable
  --> src/test/ui/resolve/issue-82825-mut.rs:17:5
   |
17 |     x.static_call();
   |     ^
   |     |
   |     cannot borrow as mutable
   |     try removing `&mut` here

error: aborting due to previous error
```

What do you think?

Reply:
oh hmm, yeah we would likely need to improve this diagnostic

Reply:
Oh also we should make sure we can still call it with `(x).static_call()` so that this is backwards compatible